crates/bevy_pbr/src/light.rs

```diff
@@ -204,6 +203,113 @@ pub enum SimulationLightSystems {
 // The z-slicing method mentioned in the aortiz article is originally from Tiago Sousa’s Siggraph 2016 talk about Doom 2016:
 // http://advances.realtimerendering.com/s2016/Siggraph2016_idTech6.pdf
 
+/// Configure the far z-plane mode used for the furthest depth slice for clustered forward
+/// rendering
+#[derive(Debug, Copy, Clone)]
+pub enum ClusterFarZMode {
+    /// Use the camera far-plane to determine the z-depth of the furthest cluster layer
+    CameraFarPlane,
+    /// Calculate the required maximum z-depth based on currently visible lights.
+    /// Makes better use of available clusters, speeding up GPU lighting operations
+    /// at the expense of some CPU time and using more indices in the cluster light
+    /// index lists.
+    MaxLightRange,
+    /// Constant max z-depth
+    Constant(f32),
+}
+
+/// Configure the depth-slicing strategy for clustered forward rendering
+#[derive(Debug, Copy, Clone)]
+pub struct ClusterZConfig {
+    /// Far z plane of the first depth slice
+    pub first_slice_depth: f32,
+    /// Strategy for how to evaluate the far z plane of the furthest depth slice
+    pub far_z_mode: ClusterFarZMode,
+}
+
+impl Default for ClusterZConfig {
+    fn default() -> Self {
+        Self {
+            first_slice_depth: 5.0,
+            far_z_mode: ClusterFarZMode::MaxLightRange,
+        }
+    }
+}
+
+/// Configuration of the clustering strategy for clustered forward rendering
+#[derive(Debug, Copy, Clone, Component)]
+pub enum ClusterConfig {
+    /// Disable light cluster calculations for this view
+    None,
+    /// One single cluster. Optimal for low-light complexity scenes or scenes where
+    /// most lights affect the entire scene.
+    Single,
+    /// Explicit x, y and z counts (may yield non-square x/y clusters depending on the aspect ratio)
+    XYZ {
+        dimensions: UVec3,
+        z_config: ClusterZConfig,
+    },
+    /// Fixed number of z slices, x and y calculated to give square clusters
+    /// with at most total clusters
```

Comment:
```suggestion
    /// Fixed number of z slices, x and y calculated to give square clusters
    /// with at most total clusters. For top-down games where lights will generally always be within a
    /// short depth range, it may be useful to use this configuration with 1 or few z slices. This
    /// would reduce the number of lights per cluster by distributing more clusters in screen space
    /// x/y which matches how lights are distributed in the scene.
```